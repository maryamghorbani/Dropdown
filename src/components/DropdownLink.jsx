import { useDispatch, useSelector } from "react-redux";
import { changeVisibility } from '../redux/dropdown/dropDownAction';


const DropdownLink = () => { 
  const dispatch = useDispatch();
  const companyName = useSelector((state) => state.company.selectedCompany);

  const onDropDown = () => {
    dispatch(changeVisibility())
  }

  return(
  <div className="nav__link" onClick={onDropDown}>
    <div className="nav__link-text-wrapper">
      <div className="nav__link-text">
        Elon Musk
      </div>
      <div className="nav__link-subtext">
        {companyName}
      </div>
    </div>

    <i className="material-icons-outlined nav__link-icon">
      settings
    </i>
  </div>
  )
  }

export default DropdownLink