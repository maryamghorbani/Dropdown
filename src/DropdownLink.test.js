import { shallow } from 'enzyme'
import DropdownLink from './components/DropdownLink'

describe('<DropdownLink />', () => {
  it('renders link', () => {
    expect(shallow(<DropdownLink />)).toMatchSnapshot()
  })
})