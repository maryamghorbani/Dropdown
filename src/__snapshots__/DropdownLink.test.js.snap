// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownLink /> renders link 1`] = `
<div
  className="nav__link"
>
  <div
    className="nav__link-text-wrapper"
  >
    <div
      className="nav__link-text"
    >
      Elon Musk
    </div>
    <div
      className="nav__link-subtext"
    >
      Viljatootja AS
    </div>
  </div>
  <i
    className="material-icons-outlined nav__link-icon"
  >
    settings
  </i>
</div>
`;
